apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useQuery } from '@tanstack/react-query';
+
+import { truncate } from '../../utils/stringUtils';
+import { TxTimeType } from '../tx-time/TxTimeType';
+
+import type {
+    SuiEventEnvelope,
+    PaginatedEvents,
+    SuiEvents,
+} from '@mysten/sui.js';
+
+import { useRpc } from '~/hooks/useRpc';
+import { Banner } from '~/ui/Banner';
+import { Link } from '~/ui/Link';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { TableCard } from '~/ui/TableCard';
+
+const TRANSACTION_STALE_TIME = 10 * 1000;
+const TRUNCATE_LENGTH = 16;
+
+export function RecentModulesCard() {
+    const rpc = useRpc();
+
+    const columns = [
+        {
+            headerLabel: 'Time',
+            accessorKey: 'time',
+        },
+        {
+            headerLabel: 'Package ID',
+            accessorKey: 'packageId',
+        },
+        {
+            headerLabel: 'Transaction ID',
+            accessorKey: 'txnDigest',
+        },
+        {
+            headerLabel: 'Sender',
+            accessorKey: 'sender',
+        },
+    ];
+
+    const transformTable = (events: SuiEvents) => ({
+        data: events.map((resp: SuiEventEnvelope) => {
+            const { event, timestamp, txDigest } = resp;
+            return {
+                ...('publish' in event && {
+                    time: <TxTimeType timestamp={timestamp} />,
+
+                    sender: (
+                        <Link
```

Comment:
I'd be open to a new `InternalLink` (probably need a better name) that took a `type` like address, object, etc and linked there, and also managed consistent truncation. Basically `LongText` but rebuilt on new UI components.

You don't need to do this now, we can create a backlog item for it.

Reply:
Agreed, we can just build that out later.